### DISCUSSION TITLE
Need help troubleshooting a script that hangs on large folders

### DISCUSSION THREAD

**Discussion by yerlaser** [nushell/nushell#6319] [Q&A]
I have written a script to search (currently using ripgrep) in all files in the current folder or list of files passed via a pipeline.
I am also saving the result in an env to be able to access it later.
The script works as expected on a small folder, but on the project for which I've written the script for it hangs.
There are several thousands of text and CSV files in it and the problem is not that it's just slow, I left it running for a day.
Raw Ripgrep processes it within a minute.
I also tried removing par- from par-each, but it doesn't help.

```
# Filter piped list for files that contain search pattern
def-env pgrep [
  --after_context (-A): int = 7 # Number of lines to show after each match (default 7)
  --before_context (-B): int = 3 # Number of lines to show before each match (default 3)
  --fixed_string (-F) # Treat pattern as fixed string
  search_pattern: string # Pattern to search
] {
  let input = $in
  let input = (if $input == null {ls -a **/*} else {$input})
  let coln = 'found_' + ($search_pattern | str replace -a '[^a-zA-Z0-9]' '_')
  let pres = (
    $input | where type == file | par-each {
      |it| $it | insert $coln (
        if $fixed_string {
          rg -F -A $after_context -B $before_context -n $search_pattern $it.name | lines
        } else {
          rg -A $after_context -B $before_context -n $search_pattern $it.name | lines
        }
      )
    } | where ($it | get $coln | length) > 0
  )
  let-env LAST_CMD_RESULT = (
    if ($pres | length) > 0 {$pres | sort-by name} else {null}
  )
  $env.LAST_CMD_RESULT
}

# Get last command result
def l [] {
  $env.LAST_CMD_RESULT
}
```

**Comment by fdncred** [comment]
This type of thing is tough to debug just by looking at code here. We would probably need to be able to reproduce the problem to see where the issue lies. It sounds like this would be a good benchmark for use to test with.

**Comment by kubouch** [comment]
I have no idea. I think you'd have to break it down and isolate the thing that causes it to hang. Maybe it's only some file being present that causes it to hang? It's hard to tell where the issue could be just looking at the full code.

**Comment by fdncred** [comment]
i ran your script on windows and it just started spewing hits real fast.
```
pgrep --fixed_string test
```
then i got a bunch of errors like this 
```
.git\hooks\pre-merge-commit.sample: The filename, directory name, or volume label syntax is incorrect. (os error 123)
.git\config: The filename, directory name, or volume label syntax is incorrect. (os error 123)
.git\description: The filename, directory name, or volume label syntax is incorrect. (os error 123)
.git\COMMIT_EDITMSG: The filename, directory name, or volume label syntax is incorrect. (os error 123)
.git\hooks\commit-msg.sample.cargo\config.toml: : The filename, directory name, or volume label syntax is incorrect.The filename, directory name, or volume label syntax is incorrect. (os error  (os error 123123))
```
maybe parameters are not being expanded right or maybe this is what is expected for a .git folder?

**Reply by yerlaser** [reply]
Hmm, interesting.
OK. Thank you very much for the trials.
I am trying currently to do the other way around, i.e. use `rg` output and then attached details to it.
But, I will try your suggestions.
Thanks again!
